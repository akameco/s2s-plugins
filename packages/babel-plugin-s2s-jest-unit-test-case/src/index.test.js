// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const getFixturesPath = x => path.resolve(__dirname, '__fixtures__', x)

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  pluginOptions: { from: getFixturesPath('export-sum.js') },
  tests: [
    `// empty`,
    {
      title: 'when exist test("sum")',
      code: `
    test('sum', () => {
      expect(sum(1, 1)).toBe(2)
    })
    `,
    },
  ],
})

pluginTester({
  title: 'no export',
  plugin,
  snapshot: false,
  pluginOptions: { from: getFixturesPath('no-export-sum.js') },
  tests: [`// empty`],
})

pluginTester({
  plugin,
  tests: [
    {
      title: 'throw error',
      code: `// throw error`,
      error: /required from option/,
    },
  ],
})
