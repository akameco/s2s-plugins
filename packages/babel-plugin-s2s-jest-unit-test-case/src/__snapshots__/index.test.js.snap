// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. s2s-jest-unit-test-case 1`] = `
"
// empty

      ↓ ↓ ↓ ↓ ↓ ↓

// empty
test(\\"sum\\", () => {
  const result = sum();
  expect(result).toBe(null);
});
"
`;

exports[`when exist test("sum") 1`] = `
"
test('sum', () => {
  expect(sum(1, 1)).toBe(2)
})

      ↓ ↓ ↓ ↓ ↓ ↓

test('sum', () => {
  expect(sum(1, 1)).toBe(2);
});
"
`;
